src/components/ModalWindow/ModalWindow.tsx

```diff
@@ -1,44 +1,59 @@
-import React, { FC, useState, FormEvent, ChangeEvent } from "react";
+import React, { FC, useState, useLayoutEffect, FormEvent, ChangeEvent } from "react";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Modal from "react-bootstrap/Modal";
-import { useAppDispatch } from "../../hooks/hooks";
-import { addModalTodo } from "../../store/reducers/UserSlice";
+import { IChange, TodoData } from "../../models/ITodo";
+import { useAppDispatch, useAppSelector } from "../../hooks/hooks";
+import { addModalTodo, updateTodo } from "../../store/reducers/UserSlice";
 import s from "./ModalWindow.module.scss";
 
-const ModalWindow: FC = () => {
-  type TodoData = {
-    message: string;
-    date: string;
-    expDate: string;
-  };
-
-  const [input, setInput] = useState<TodoData>({ message: "", date: "", expDate: "" });
+const ModalWindow: FC<IChange> = ({ type, message, dateStored, idd }) => {
+  const [input, setInput] = useState<TodoData>({ message: "", date: "", expDate: "", idd: `${idd}` });
   const [show, setShow] = useState(false);
 
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
 
   const dispatch = useAppDispatch();
 
+  const todos = useAppSelector((state) => state.todos.todos);
+
   const handleSubmit = (e: FormEvent<HTMLButtonElement>) => {
     e.preventDefault();
     if (/^[A-Za-zА-Яа-яЁё0-9\s]+$/.test(input.message) && /[0-9]+/.test(input.date) && /[0-9]+/.test(input.expDate)) {
-      dispatch(addModalTodo(input));
-      setInput({ message: "", date: "", expDate: "" });
-      setShow(false);
+      if (!type) {
+        dispatch(addModalTodo(input));
+        setInput({ message: "", date: "", expDate: "", idd: "" });
+        setShow(false);
+      } else {
+        dispatch(updateTodo(input));
+        setShow(false);
+      }
     }
   };
-
   const handleChange = (e: ChangeEvent<HTMLInputElement>, fieldName: string): void => {
     setInput({ ...input, [fieldName]: e.target.value });
   };
 
+  useLayoutEffect(() => {
+    setInput({ ...input, message: `${message}`, date: `${dateStored?.date}`, expDate: `${dateStored?.expDate}` });
+    console.log(new Date());
+    console.log(new Date().toLocaleDateString());
+    console.log(new Date().toLocaleTimeString());
+    console.log(new Date().getTime());
+    console.log(new Date().getHours());
+    console.log(new Date().getMinutes());
+  }, [type]);
+
   return (
     <>
       <Form>
         <span onClick={handleShow}>
-          <i className={`fa fa-plus ${s.plusSimbol}`} aria-hidden="true"></i>
+          {type ? (
```

Comment:
I prefer to add condition inside className, because the difference is in the classes only

Reply:
Good idea, edits made